lib/gel/gemfile_parser.rb

```diff
@@ -38,26 +52,30 @@ def git_source(name, &block)
       @result.git_sources[name] = block
     end
 
-    def ruby(version, engine: nil, engine_version: nil)
-      req = Gel::Support::GemRequirement.new(version)
-      unless req.satisfied_by?(Gel::Support::GemVersion.new(RUBY_VERSION))
+    def ruby(*versions, engine: nil, engine_version: nil)
+      req = Gel::Support::GemRequirement.new(versions)
+      running_ruby_version = RunningRuby.version
+      running_engine = RunningRuby.engine
+      running_engine_version = RunningRuby.engine_version
+
+      unless req.satisfied_by?(Gel::Support::GemVersion.new(running_ruby_version))
         raise Gel::Error::MismatchRubyVersionError.new(
-          running: RUBY_VERSION,
-          requested: version,
+          running: running_ruby_version,
+          requested: versions,
         )
       end
-      unless !engine || RUBY_ENGINE == engine
+      unless !engine || running_engine == engine
         raise Gel::Error::MismatchRubyEngineError.new(
-          running: RUBY_ENGINE,
+          running: running_engine,
           engine: engine,
         )
       end
       if engine_version
+        req = Gel::Support::GemRequirement.new(engine_version)
         raise "Cannot specify :engine_version without :engine" unless engine
-        req = Gel::Support::GemRequirement.new(version)
```

Comment:
Here we should `Gel::Support::GemRequirement.new` with `engine_version` instead 🙈